staging/kos/pkg/controllers/ipam/ipam.go

```diff
@@ -847,7 +844,7 @@ func (ctlr *IPAMController) clearNetworkAttachmentData(ns, name string) {
 
 func AttachmentSubnets(obj interface{}) (subnets []string, err error) {
 	att := obj.(*netv1a1.NetworkAttachment)
-	return []string{att.Spec.Subnet}, nil
+	return []string{att.Namespace + "/" + att.Spec.Subnet}, nil
```

Comment:
This avoids unnecessary work: when a notification about a subnet X is received  the indexer is used to retrieve all the attachments in X and enqueue them. Without this change, the attachments in subnets with the same name but different namespaces as X are also enqueued

Reply:
yes